backend/test/modules/doubles-tournament/core/application/BookTableAndPublishEventAfterCallingMatch.spec.ts

```diff
@@ -0,0 +1,187 @@
+
+  it('Then publish event and execute BookTournamentTable command from different module | everything success', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.success();
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResultFromCallingMatch = await doublesTournament.executeCommand(callMatch);
+
+    const bookTable = new BookTournamentTable(tournamentId, tableNumber);
+    const commandResultFromBookingTable = await doublesTournament.executeCommand(bookTable);
+
+    //Then
+    expect(commandResultFromCallingMatch.isSuccess()).toBeTruthy();
+    expect(commandResultFromBookingTable.isSuccess()).toBeTruthy();
+
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(3);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  it('Then publish event and execute BookTournamentTable command from different module | table has been already booked', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.failureDueTo(new Error('Table has been already booked!'));
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResultFromCallingMatch = await doublesTournament.executeCommand(callMatch);
+
+    const bookTable = new BookTournamentTable(tournamentId, tableNumber);
```

Comment:
Tego nie rozumiem, dlaczego gdzieś z boku robisz book table? Przecież to i tak rzeczywiscie sie nie wykonuje. 
Tzn. teraz testujesz swojego mocka. 

To raczej powinno byc preztestowane w module zajmowania stolu - a Ty tylko juz testujesz z zalozeniem, ze to failuje - bo to wlasnie napisales w GIVEN, ze stol jest zajety poprzez handlera. To moze fajnie zrobic metode np `tableAlreadyBooked`, ktora wlasnie doda nam commandhandler failujacy dla danego id stolu :) ? I wtedy widać w tescie tez o co chodzi z handlerem.

Reply:
![image](https://user-images.githubusercontent.com/31566345/111843575-bd5ea080-8901-11eb-963a-c8b73c6eb15b.png)
no właśnie wg mnie też ta daleko poleciałem z tym i chyba wystarczy tylko tej jeden test nad zakomentowaną kreską?
Czy jest sens dalej walczyć z jakąś metodą `tableAlreadyBooked` skoro to już jest chyba trochę poza zakresem taska?

Reply:
Jest sens, ale nie do konca dobrze do tego podszedłeś, to co odpisałem gdzieś wyżej powtarzam:
Twój test powinien sprawdzić dwa przypadki. Jak zareaguje Twój command handler, jeśli:
1. Book się uda
2. Book się nie uda (bo np. już został zabukowany na coś innego)

Chyba proste, nie :) ?

Reply:
Tak racja, trochę inaczej to zrozumiałem, ale jak napisałeś to już jasne.
Dla pewności - o to chodziło?
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/pull/74/commits/83bbba09cbcbe310eb371ddda662b1589ecfdaf6

Reply:
Ooo, dokładnie :) 
Tylko czy ten 
```
if (!bookTable.isSuccess()) {
      return CommandResult.failureDueTo(new Error('Table has been already booked!'));
    }
```

if jest potrzebny?

Reply:
a.AAA, że wgl to nie jest potrzebne, bo i tak jak to
`await this.commandPublisher.execute(new BookTournamentTable(command.tournamentId, command.tableNumber));`
zwróci nam failure to execute na komendzie CallMatch też nie dojdzie już do końca?

Reply:
no dobra sprawdziłem w testach i jednak CallMatch wtedy i tak dojdzie do końca

To co masz na myśli w takim razie czy ten if jest nam potrzebny?

Reply:
Bo skoro nie uda się zarezerwować stołu, no to powinna być blokada z wywołaniem meczu - logiczne

więc:
1. tak ten if jest potrzebny
2. jedynie sposób jego zapisu może być inny
?